aten/src/ATen/core/dispatch/OperatorEntry.h

```diff
@@ -173,11 +173,8 @@ class TORCH_API OperatorEntry final {
 
   [[noreturn]] void reportError(DispatchKey dispatchKey) const;
 
-  const KernelFunction& lookup(DispatchKey k) const {
-    const auto idx = getDispatchTableIndexForDispatchKey(k);
-    if (C10_UNLIKELY(idx == -1)) {
```

Comment:
So, I found that this -1 check was added for mobile [here](https://github.com/pytorch/pytorch/pull/65732/files#diff-2bedde63e4afa79382ec696bccac8a87777cd095453e859e5f60fae07f588a6cR303) (cc @dhruvbird), but I'm not sure that it's necessary.

Instead, in this PR, I have `getDispatchTableIndexForDispatchKeySet()` return 0 when it finds an invalid key, which corresponds to the `Undefined` kernel - which should raise an error when it gets called, in the same way that we're explicitly trying to raise the error here.

Reply:
Thanks for mentioning this @bdhirsh - seems reasonable! Maybe the only difference would be that the check with `-1` would be compiled away for server builds, whereas the check for `0` won't, and will get hit at runtime. cc @swolchok who did this analysis earlier.

Reply:
Because DispatchKey has an underlying type (uint8_t), I don't think that stuffing -1 in there would be UB, and therefore IU don't think that the check for -1 could be compiled away in server builds.

Reply:
Good call. I missed the fact that `getDispatchTableIndexForDispatchKeySet` now returns `uint64_t`.